velox/exec/RowContainer.h

```diff
@@ -178,6 +178,68 @@ class RowColumn {
   const uint64_t packedOffsets_;
 };
 
+/// Aggregated stats of a column in the 'RowContainer'.
+class RowColumnStats {
+ public:
+  RowColumnStats() = default;
+
+  void addCellSize(int32_t bytes) {
+    if (UNLIKELY(nonNullCount_ == 0)) {
+      minBytes_ = bytes;
+      maxBytes_ = bytes;
+    } else {
+      minBytes_ = std::min(minBytes_, bytes);
+      maxBytes_ = std::max(maxBytes_, bytes);
+    }
+    sumBytes_ += bytes;
+    ++nonNullCount_;
+  }
+
+  void addNullCell() {
+    ++nullCount_;
+  }
+
+  int32_t maxBytes() const {
+    return maxBytes_;
+  }
+
+  int32_t minBytes() const {
+    return minBytes_;
+  }
+
+  uint64_t sumBytes() const {
+    return sumBytes_;
+  }
+
+  uint64_t avgBytes() const {
+    if (nonNullCount_ == 0) {
+      return 0;
+    }
+    return sumBytes_ / nonNullCount_;
+  }
+
+  uint32_t nonNullCount() const {
+    return nonNullCount_;
+  }
+
+  uint32_t nullCount() const {
+    return nullCount_;
+  }
+
+  /// Merges multiple aggregated stats of the same column into a single one.
+  static RowColumnStats mergeStats(
+      const std::vector<RowColumnStats>& statsList);
+
+ private:
+  // Aggregated stats for non-null rows of the column.
+  int32_t minBytes_{0};
```

Comment:
why not use uint32_t? Thanks!

Reply:
int32_t is widely used as cell size in row container. I'm trying to make it compatible so that no cast is needed when performing ::max ::min. If we want we can have another PR for refactoring the types in general.